### PR TITLE
Handle TF error messages

### PR DESCRIPTION
**Pull request by benrudolph**
@snopoke @wpride this better handles errors so it doesn't falsely say "success"

### CODE REVIEW

File: corehq/apps/cloudcare/static/cloudcare/js/backbone/apps.js [hunk 1]
```diff
@@ -899,8 +899,16 @@ hqDefine('cloudcare/js/backbone/apps.js', function () {
             $('#sync-button').disableButton();
             showLoading();
             resp.done(function (data) {
-                tfSyncComplete(data.status === "error");
+                var hasError = data.status === "error"
```
Comment (dimagimon) [inline corehq/apps/cloudcare/static/cloudcare/js/backbone/apps.js]: Missing semicolon. (semi)

File: corehq/apps/cloudcare/static/cloudcare/js/backbone/apps.js [hunk 2]
```diff
@@ -899,8 +899,16 @@ hqDefine('cloudcare/js/backbone/apps.js', function () {
...
+                tfSyncComplete(hasError);
+                if (hasError) {
+                    console.error(data.message);
+                }
                 $('#sync-button').enableButton();
+            })
+            resp.error(function(data) {
+                tfSyncComplete(true);
+                console.error(data.responseJSON);
```
Comment (dimagimon) [inline corehq/apps/cloudcare/static/cloudcare/js/backbone/apps.js]: Unexpected console statement. (no-console)

File: corehq/apps/cloudcare/static/cloudcare/js/backbone/apps.js [hunk 3]
```diff
@@ -899,8 +899,16 @@ hqDefine('cloudcare/js/backbone/apps.js', function () {
...
             showLoading();
             resp.done(function (data) {
-                tfSyncComplete(data.status === "error");
+                var hasError = data.status === "error"
+                tfSyncComplete(hasError);
+                if (hasError) {
+                    console.error(data.message);
+                }
                 $('#sync-button').enableButton();
+            })
```
Comment (dimagimon) [inline corehq/apps/cloudcare/static/cloudcare/js/backbone/apps.js]: Missing semicolon. (semi)

File: corehq/apps/cloudcare/static/cloudcare/js/backbone/apps.js [hunk 4]
```diff
@@ -899,8 +899,16 @@ hqDefine('cloudcare/js/backbone/apps.js', function () {
             $('#sync-button').disableButton();
             showLoading();
             resp.done(function (data) {
-                tfSyncComplete(data.status === "error");
+                var hasError = data.status === "error"
+                tfSyncComplete(hasError);
+                if (hasError) {
+                    console.error(data.message);
```
Comment (dimagimon) [inline corehq/apps/cloudcare/static/cloudcare/js/backbone/apps.js]: Unexpected console statement. (no-console)

File: corehq/apps/cloudcare/static/cloudcare/js/backbone/apps.js [hunk 5]
```diff
@@ -899,9 +899,17 @@ hqDefine('cloudcare/js/backbone/apps.js', function () {
...
+                var hasError = data.status === "error";
+                tfSyncComplete(hasError);
+                if (hasError) {
+                    console.error(data.message);
+                }
                 $('#sync-button').enableButton();
             });
+            resp.error(function(data) {
+                tfSyncComplete(true);
+                console.error(data.responseJSON);
```
Comment (dimagimon) [inline corehq/apps/cloudcare/static/cloudcare/js/backbone/apps.js]: Unexpected console statement. (no-console)

File: corehq/apps/cloudcare/static/cloudcare/js/backbone/apps.js [hunk 6]
```diff
@@ -899,9 +899,17 @@ hqDefine('cloudcare/js/backbone/apps.js', function () {
             $('#sync-button').disableButton();
             showLoading();
             resp.done(function (data) {
-                tfSyncComplete(data.status === "error");
+                var hasError = data.status === "error";
+                tfSyncComplete(hasError);
+                if (hasError) {
+                    console.error(data.message);
```
Comment (dimagimon) [inline corehq/apps/cloudcare/static/cloudcare/js/backbone/apps.js]: Unexpected console statement. (no-console)

File: corehq/apps/cloudcare/static/cloudcare/js/backbone/apps.js (hunk 5 above)
Comment (benrudolph) [inline corehq/apps/cloudcare/static/cloudcare/js/backbone/apps.js]: (ignoring)

**Comment by benrudolph** [comment]
just a reminder we probably should wait for tests on these

**Pull request merged**